File: Java/RetrofitApps/HeroRetofitApp/app/src/main/java/com/e/vasialeleka/heroretofitapp/MainActivity.java
package com.e.vasialeleka.heroretofitapp;

import android.annotation.SuppressLint;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.util.Log;
import android.widget.Toast;

import java.util.List;

import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;
import retrofit2.Retrofit;
import retrofit2.converter.gson.GsonConverterFactory;

public class MainActivity extends AppCompatActivity {
RecyclerView recyclerView;
Adapter adapter;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        recyclerView = findViewById(R.id.recView);

       LinearLayoutManager layoutManager = new LinearLayoutManager(MainActivity.this);
        recyclerView.setLayoutManager(layoutManager);
        Retrofit retrofit = new Retrofit.Builder()
                .baseUrl(Api.BaseUrl)
                .addConverterFactory(GsonConverterFactory.create())
                .build();
        Api api = retrofit.create(Api.class);

        Call<List<Hero>> call  =  api.getHeroes();

        call.enqueue(new Callback<List<Hero>>() {
            @SuppressLint("LongLogTag")
            @Override
            public void onResponse(Call<List<Hero>> call, Response<List<Hero>> response) {
                List<Hero> heroes = response.body();
                adapter = new Adapter(heroes,MainActivity.this);
                recyclerView.setAdapter(adapter);
             adapter.notifyDataSetChanged();
            }

            @Override
            public void onFailure(Call<List<Hero>> call, Throwable t) {
                Toast.makeText(getApplicationContext(),t.getMessage(),Toast.LENGTH_SHORT).show();

            }
        });
    }
}

File: Kotlin/Rerorofit/app/src/main/java/com/example/vasialeleka/newsappkotlin/MainActivity.kt
package com.example.vasialeleka.newsappkotlin

import android.annotation.SuppressLint
import android.support.v7.app.AppCompatActivity
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager

import android.support.v7.widget.RecyclerView
import android.widget.Toast
import retrofit2.*
import retrofit2.converter.gson.GsonConverterFactory


class MainActivity : AppCompatActivity() {
// val API_KEY = "e60697f7f9874f35a82708c0df1618a9"
  //  val BASE_URL = ""
lateinit var recView : RecyclerView
lateinit var adapter :  Adapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        recView = findViewById(R.id.recView)

        var linearLayoutManager = LinearLayoutManager(this)
        recView.layoutManager = linearLayoutManager

        var retrofit = Retrofit.Builder()
            .baseUrl(Api.BASE_URL)
            .addConverterFactory(GsonConverterFactory.create())
            .build()
        var api = retrofit.create(Api::class.java)
        var call = api.heroes

        call.enqueue(object : Callback<List<Hero>> {
            @SuppressLint("LongLogTag")
            override fun onResponse(call: Call<List<Hero>>, response: Response<List<Hero>>) {
                val heroes = response.body()
                adapter = Adapter(heroes!!, this@MainActivity)
                recView.adapter = adapter
                adapter.notifyDataSetChanged()
            }

            override fun onFailure(call: Call<List<Hero>>, t: Throwable) {
                Toast.makeText(applicationContext, t.message, Toast.LENGTH_SHORT).show()

            }
        })




    }
}
